main.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python
+# docker run -it --rm --env-file .env -v /path/to/folder/atd-traffic-incident-reports:/app \
+# atddocker/atd-traffic-incident-reports:production python main.py
+
+import os
+import logging
+import sys
+import requests
+import cx_Oracle
+import hashlib
+import arrow
+
+
+USER = os.getenv("USER")
+PASSWORD = os.getenv("PASSWORD")
+HOST = os.getenv("HOST")
+PORT = os.getenv("PORT")
+SERVICE = os.getenv("SERVICE")
+PGREST_ENDPOINT = os.getenv("PGREST_ENDPOINT")
+PGREST_TOKEN = os.getenv("PGREST_TOKEN")
+
+headers = {
+    "Content-Type": "application/json",
+    "Authorization": f"Bearer {PGREST_TOKEN}",
+    "Prefer": "return=representation, resolution=merge-duplicates",
+}
+
+QUERY = "SELECT * FROM QACT_USER.QACT"
+
+
+def get_conn(host, port, service, user, password):
+    """ Create oracle database connection """
+    # make connect descriptor string
+    oracle_data_source_name = cx_Oracle.makedsn(host, port, service_name=service)
+    # create and return oracle connection object
+    return cx_Oracle.connect(user=user, password=password, dsn=oracle_data_source_name)
+
+
+def get_oracle_db_records():
+    """
+    connects to oracle db and executes query, converts rows retrieved into dicts
+    :return: list of records (dict)
+    """
+    conn = get_conn(HOST, PORT, SERVICE, USER, PASSWORD)
+    cursor = conn.cursor()
+    cursor.execute(QUERY)
+
+    # define row handler which returns each row as a dict
+    # h/t https://stackoverflow.com/questions/35045879/cx-oracle-how-can-i-receive-each-row-as-a-dictionary
+    cursor.rowfactory = lambda *args: dict(
+        zip([d[0] for d in cursor.description], args)
+    )
+    rows = cursor.fetchall()
+    conn.close()
+
+    logging.info(f"{len(rows)} records in oracle database received.")
+    return rows
+
+
+def generate_record_id(call_number, timestamp):
+    """
+    the records from the oracle database do not have record ids associated with them
+    the coldfusion app added IDs generated by Coldfusion using #hash(call_number,"SHA")#
+    this function does the same using hashlib and appending our timestamp as before
+    :param call_number: call_number
+    :param timestamp:
+    :return: traffic_incident_id string
+    """
+    # create coldfusion hash id
+    cf_id = hashlib.sha1(str.encode(call_number)).hexdigest().upper()
```

Comment:
Thanks again for finding out how to do this - that would have been rough to reverse engineer.